src/decision/assertion_list.h

```diff
@@ -0,0 +1,105 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * Assertion list
+ */
+
+#include "cvc5_private.h"
+
+#ifndef CVC5__DECISION__ASSERTION_LIST_H
+#define CVC5__DECISION__ASSERTION_LIST_H
+
+#include <iosfwd>
+
+#include "context/cdlist.h"
+#include "context/cdo.h"
+#include "expr/node.h"
+
+namespace cvc5 {
+namespace decision {
+
+/**
+ * For monitoring activity of assertions
+ */
+enum class DecisionStatus
+{
+  // not currently watching status of the current assertion
+  INACTIVE,
+  // no decision was made considering the assertion
+  NO_DECISION,
+  // a decision was made considering the assertion
+  DECISION,
+  // we backtracked while considering the assertion
+  BACKTRACK
+};
+const char* toString(DecisionStatus s);
+std::ostream& operator<<(std::ostream& out, DecisionStatus s);
+
+/**
+ * An assertion list used by the justification heuristic. This tracks a list
+ * of formulas that we must justify.
+ */
+class AssertionList
+{
+ public:
+  /**
+   * @param ac The context on which the assertions depends on. This is the
+   * user context for assertions. It is the SAT context for assertions that
+   * are dynamically relevant based on what is asserted, e.g. lemmas
+   * corresponding to skolem definitions.
+   * @param ic The context on which the current index of the assertions
+   * depends on. This is typically the SAT context.
+   * @param dyn Whether to use a dynamic ordering of the assertions. If this
+   * flag is true, then getNextAssertion will return the most important next
+   * assertion to consider based on heuristics in response to notifyStatus.
+   */
+  AssertionList(context::Context* ac,
+                context::Context* ic,
+                bool useDyn = false);
+  virtual ~AssertionList() {}
+  /** Presolve, which clears the dynamic assertion order */
+  void presolve();
+  /** Add the assertion n */
+  void addAssertion(TNode n);
+  /**
+   * Get the new assertion, increment d_assertionIndex, sets fromIndex to the
```

Comment:
Why does a local variable `fromIndex` matter here? Maybe add some more documentation to clarify.

Reply:
This was actually leftover from something else I was trying and is not needed anymore, I'll remove.